gli/utils.py

```diff
@@ -392,3 +433,27 @@ def save_data(prefix, **kwargs):
     for key, matrix in sparse_arrays.items():
         sp.save_npz(f"{prefix}_{key}.sparse.npz", matrix)
         print("Save sparse matrix", key, "to", f"{prefix}_{key}.sparse.npz")
+
+
+def get_local_data_dir():
+    """Get the local data storage directory.
+
+    The function is used to determine whether the gli module is cloned from
+    GitHub or downloaded by pypi. If the module is cloned from GitHub, the
+    local data storage directory is ``./datasets``. Otherwise, the local data
+    storage directory is ``~/.gli/datasets``.
+
+    :return: gli's local data storage directory
+    :rtype: str
+    """
+    # The repo is cloned to the local file system and is complete
```

Comment:
There is also a case where a dataset contributor clones the dataset with partial checkout. `./datasets` folder exists but the datasets are incomplete.

Do we need to have special treatment for this case? In any case, it may worth revising the comment here to recognize this corner case.

Reply:
I am not sure how partial checkout works. But I will add this piece in the docstring and see what we should do when we implement that feature later.